src/mount/notification_area_logging.h

```diff
@@ -0,0 +1,170 @@
+/*
+ Copyright 2023      Leil Storage OÜ
+
+ SaunaFS is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, version 3.
+
+ SaunaFS is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with SaunaFS  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "common/platform.h"
+
+#ifdef _WIN32
+#include <shellapi.h>
+#include <windows.h>
+#include <string>
+#else
+#include <cstdlib>
+#endif
+
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <unordered_map>
+
+inline bool gShowMessagesOnNotificationArea = false;
+
+struct MessageCache {
+	std::string message;
+	std::chrono::steady_clock::time_point timestamp;
+};
+
+inline std::jthread notificationThread;
+
+inline std::unordered_map<std::string, MessageCache> messageCache;
+inline unsigned gMessageSuppressionPeriod;
+inline std::mutex messageCacheMutex;
+inline std::condition_variable cv;
+inline std::atomic<bool> stopThread = false;
+
+#ifdef _WIN32
+inline void ShowWindowsNotification(const std::string &message) {
+	// Create a hidden window
+	HWND hWnd = CreateWindowEx(0, "STATIC", "HiddenWindow", 0, 0, 0, 0, 0,
+	                           HWND_MESSAGE, NULL, NULL, NULL);
+
+	// Define the NOTIFYICONDATA structure
+	NOTIFYICONDATA nid = {};
+	nid.cbSize = sizeof(NOTIFYICONDATA);
+	nid.hWnd = hWnd;                   // Handle to the hidden window
+	nid.uID = 1001;                    // Unique ID for the notification icon
+	nid.uFlags = NIF_INFO | NIF_ICON;  // Display an info balloon and an icon
+	nid.dwInfoFlags = NIIF_INFO;       // Info icon
+	strcpy_s(nid.szInfo, message.c_str());                // Set the message
+	strcpy_s(nid.szInfoTitle, "SaunaFS Windows Client");  // Set the title
+	nid.hIcon = LoadIcon(NULL, IDI_INFORMATION);  // Load a standard icon
+
+	// Display the notification
+	if (!Shell_NotifyIcon(NIM_ADD, &nid)) {
+		// Handle error
+		safs::log_debug(
+		    "ShowMessageOnWindowsNotificationArea: Failed to "
+		    "display notification icon");
+	}
+	if (!Shell_NotifyIcon(NIM_MODIFY, &nid)) {
+		// Handle error
+		safs::log_debug(
+		    "ShowMessageOnWindowsNotificationArea: Failed to "
+		    "modify notification icon");
+	}
+
+	// Clean up
+	DestroyWindow(hWnd);
+}
+#else
+inline void ShowLinuxNotification(const std::string &message) {
+	std::string command =
+	    "notify-send \"SaunaFS Linux Client\" \"" + message + "\"";
```

Comment:
Consider using a Raw String Literals to avoid escaping special characters and improve the readability:
```suggestion
	    R"(notify-send "SaunaFS Linux Client" ")" + message + "\"";
```

Reply:
Done